riptide-compression/src/main/java/org/zalando/riptide/compression/WrappingHttpOutputMessage.java

```diff
@@ -0,0 +1,44 @@
+package org.zalando.riptide.compression;
+
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpOutputMessage;
+import org.zalando.fauxpas.ThrowingUnaryOperator;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.io.OutputStream;
+
+final class WrappingHttpOutputMessage implements HttpOutputMessage, AutoCloseable {
+
+    private final HttpOutputMessage message;
+    private final ThrowingUnaryOperator<OutputStream, IOException> wrapper;
+    private OutputStream stream;
+
+    WrappingHttpOutputMessage(HttpOutputMessage message, ThrowingUnaryOperator<OutputStream, IOException> wrapper) {
```

Comment:
`@AllArgsConstructor`?

Reply:
`OutputStream stream` is not injected, but managed internally.